File: dongyimaicartweb/src/main/webapp/js/service/cartService.js
app.service('cartService',function ($http) {

    //写订单模块
    this.submit = function (entity) {
        return $http.post('../order/add.do',entity);
    }

    this.findAddrByUserId = function () {
        return $http.get('../address/findAddrByUserId.do');
    }

    this.findCartList = function () {
        return $http.get('../cart/findCartList.do');
    }

    this.addGoodsToCart = function (itemId,num) {
        return $http.get('../cart/addGoodsToCart.do?itemId='+itemId+"&num="+num);
    }

    this.sum = function (cartList) {
        var totalValue = {totalNum:0,totalMoney:0.00};

        for(var i=0;i<cartList.length;i++){
            var cart = cartList[i];
            for(var j=0;j<cart.orderItemList.length;j++){
                var item = cart.orderItemList[j];
                totalValue.totalNum += item.num;
                totalValue.totalMoney += item.totalFee;
            }
        }

        return totalValue;
    }

})

File: dongyimaicartinterface/src/main/java/com/dongyimai/cart/service/CartService.java
package com.dongyimai.cart.service;

import com.dongyimai.entity.Cart;
import com.dongyimai.pojo.TbOrderItem;

import java.util.List;

public interface CartService {

    public List<Cart> addGoodsToCart(List<Cart> cartList, Long itemId, Integer num);

    public List<Cart> findCartListFromRedis(String username);

    public void saveCartToRedis(String username, List<Cart> cartList);

    public List<Cart> mergeCartList(List<Cart> cartList1, List<Cart> cartList2);

}
